File: app/src/main/java/com/hana053/micropost/services/ServiceModule.java
package com.hana053.micropost.services;

import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@SuppressWarnings("WeakerAccess")
@Module
public class ServiceModule {

    @Provides
    @Singleton
    public HttpErrorHandler provideHttpErrorHandler(Context context) {
        return new HttpErrorHandlerImpl(context);
    }
}

File: app/src/main/java/com/hana053/micropost/services/ServiceModule.kt
package com.hana053.micropost.services

import com.github.salomonbrys.kodein.Kodein
import com.github.salomonbrys.kodein.bind
import com.github.salomonbrys.kodein.instance
import com.github.salomonbrys.kodein.singleton


fun serviceModule() = Kodein.Module {

    bind<HttpErrorHandler>() with singleton {
        HttpErrorHandlerImpl(instance(), instance())
    }

    bind<AuthTokenRepository>() with singleton {
        AuthTokenRepositoryImpl(instance())
    }

    bind<AuthService>() with singleton {
        AuthServiceImpl(instance(), instance())
    }

    bind<LoginService>() with singleton {
        LoginServiceImpl(instance(), instance(), instance(), instance())
    }

}